### PR TITLE
Regression module renames

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/methods/statgen.py [hunk 1]
```diff
@@ -394,32 +393,32 @@ def logreg(dataset, test, y, x, covariates=[], root='logreg'):
...
+    Wald       `logreg.beta`           float64 fit effect coefficient,
+                                               :math:`\hat\beta_1`
+    Wald       `logreg.standard_error` float64 estimated standard error,
+                                               :math:`\widehat{\mathrm{se}}`
+    Wald       `logreg.z_stat`         float64 Wald :math:`z`-statistic, equal to
+                                               :math:`\hat\beta_1 / \widehat{\mathrm{se}}`
+    Wald       `logreg.p_value`        float64 Wald p-value testing :math:`\beta_1 = 0`
+    LRT, Firth `logreg.beta`           float64 fit effect coefficient,
+                                               :math:`\hat\beta_1`
+    LRT, Firth `logreg.chi_squared`    float64 deviance statistic
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: use `chi_sq_stat` or `chi_squared_stat` to match `t_stat` and `z_stat`?

File: python/hail/methods/statgen.py [hunk 2]
```diff
@@ -428,16 +427,16 @@ def logreg(dataset, test, y, x, covariates=[], root='logreg'):
...
-                                                     convergence, explosion, or
-                                                     reaching the max (25 for
-                                                     Wald, LRT; 100 for Firth)
-    Wald, LRT, Firth `logreg.fit.converged`  bool    ``True`` if iteration converged
-    Wald, LRT, Firth `logreg.fit.exploded`   bool    ``True`` if iteration exploded
-    ================ ======================= ======= ===============================
+    ================ ========================= ======= ===============================
+    Test             Field                     Type    Value
+    ================ ========================= ======= ===============================
+    Wald, LRT, Firth `logreg.fit.n_iterations` int32   number of iterations until
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: Pythonic: num_iterations

File: python/hail/methods/statgen.py [hunk 3]
```diff
@@ -582,11 +581,11 @@ def logreg(dataset, test, y, x, covariates=[], root='logreg'):
            use_ml=bool,
            delta=nullable(numeric),
            sparsity_threshold=numeric,
-           n_eigs=nullable(int),
+           n_eigenvalues=nullable(int),
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: `num_eigenvectors`

File: python/hail/methods/statgen.py [hunk 4]
```diff
@@ -985,26 +987,29 @@ def lmmreg(ds, kinship_matrix, y, x, covariates=[], global_root="lmmreg_global",
...
-    sample variance (also known as the trace, or the sum of the eigenvalues).
-    For example, setting `dropped_variance_fraction` to 0.01 will use the
-    minimal number of eigenvectors to account for 99% of the sample variance.
-    Specifying both parameters will apply the more stringent (fewest
-    eigenvectors) of the two.
+    :func:`.linear_mixed_regression` can implicitly use a low-rank
+    approximation of the kinship matrix to more rapidly fit delta and the
+    statistics for each variant. The computational complexity per variant is
+    proportional to the number of eigenvectors used. This number can be
+    specified in two ways. Specify the parameter `n_eigenvalues` to use only the
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: `num_eigenvectors` here and next line

File: python/hail/methods/statgen.py [hunk 5]
```diff
@@ -1029,7 +1034,7 @@ def lmmreg(ds, kinship_matrix, y, x, covariates=[], global_root="lmmreg_global",
     sparsity_threshold : :obj:`float`
         Genotype vector sparsity at or below which to use sparse genotype
         vector in rotation (advanced).
-    n_eigs : :obj:`int`
+    n_eigenvalues : :obj:`int`
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: num_eigenvectors

File: python/hail/docs/methods/genetics.rst [hunk 6]
```diff
@@ -12,21 +12,21 @@ Genetics
     concordance
     filter_intervals
     FilterAlleles
-    grm
     hwe_normalized_pca
-    ibd
+    identity_by_descent
+    genetic_relatedness_matrix
+    realized_relatedness_matrix
```
Comment (jbloom22) [inline python/hail/docs/methods/genetics.rst]: realized_relationship_matrix

File: src/main/scala/is/hail/methods/LinearMixedRegression.scala [hunk 7]
```diff
@@ -18,8 +18,8 @@ import org.apache.commons.math3.util.FastMath
 object LinearMixedRegression {
   val schema: Type = TStruct(
     ("beta", TFloat64()),
-    ("sigma_g2", TFloat64()),
-    ("chi2", TFloat64()),
+    ("sigma_g_squared", TFloat64()),
+    ("chi_squared", TFloat64()),
```
Comment (jbloom22) [inline src/main/scala/is/hail/methods/LinearMixedRegression.scala]: `chi_sq_stat` or `chi_squared_stat`

File: src/main/scala/is/hail/methods/LinearMixedRegression.scala [hunk 8]
```diff
@@ -109,30 +110,42 @@ object LinearMixedRegression {
...
-        ("delta", TFloat64()), ("h2", TFloat64()), ("evals", TArray(TFloat64())), ("n_eigenvalues", TInt32()), ("dropped_variance_fraction", TFloat64())), rootGA)
+      TStruct(
+        ("use_ml", TBoolean()),
+        ("beta", TDict(TString(), TFloat64())),
+        ("sigma_g_squared", TFloat64()),
+        ("sigma_e_squared", TFloat64()),
+        ("delta", TFloat64()),
+        ("h_squared", TFloat64()),
+        ("eigenvalues", TArray(TFloat64())),
+        ("n_eigenvalues", TInt32()),
```
Comment (jbloom22) [inline src/main/scala/is/hail/methods/LinearMixedRegression.scala]: num_eigenvectors

File: src/main/scala/is/hail/stats/LogisticRegressionModel.scala [hunk 9]
```diff
@@ -74,7 +74,7 @@ case class WaldStats(b: DenseVector[Double], se: DenseVector[Double], z: DenseVe
 object LikelihoodRatioTest extends LogisticRegressionTest {
   val schema = TStruct(
     ("beta", TFloat64()),
-    ("chi2", TFloat64()),
+    ("chi_squared", TFloat64()),
```
Comment (jbloom22) [inline src/main/scala/is/hail/stats/LogisticRegressionModel.scala]: `chi_sq_stat` or `chi_squared_stat` here and below

File: python/hail/methods/statgen.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: We mix this all over our interface, but I think I do prefer `num`. I'll change it.

Comment (tpoterba) [reply python/hail/methods/statgen.py]: actually, I'm not so sure. Let's poll the group

File: python/hail/methods/statgen.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: good call

File: python/hail/methods/statgen.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: did some research, fine to change to num. We'll need to change the sample/variant qc annotations too

**Comment by tpoterba** [comment]
as part of this change, I replaced "num_" with "n_" everywhere in our Python API.

**Comment by tpoterba** [comment]
will add this to the style guide when I update it

**Pull request merged**